src/Views/HardwareView.vala

```diff
@@ -0,0 +1,254 @@
+                }
+            }
+        }
+
+        //Memory
+        memory = GLib.format_size (get_mem_info ());
+
+        // Graphics
+        try {
+            Process.spawn_command_line_sync ("lspci", out graphics);
+
+            if ("VGA" in graphics) { //VGA-keyword indicates graphics-line
+                string[] lines = graphics.split("\n");
+                graphics="";
+
+                foreach (var s in lines) {
+                    if ("VGA" in s || "3D" in s) {
+                        string model = get_graphics_from_string(s);
+
+                        if (graphics=="") {
+                            graphics = model;
+                        } else {
+                            graphics += "\n" + model;
+                        }
+                    }
+                }
+            }
+        } catch (Error e) {
+            warning (e.message);
+            graphics = _("Unknown");
+        }
+
+        // Hard Drive
+        var file_root = GLib.File.new_for_path ("/");
+        try {
+            var info = file_root.query_filesystem_info (GLib.FileAttribute.FILESYSTEM_SIZE, null);
+            hdd = GLib.format_size (info.get_attribute_uint64 (GLib.FileAttribute.FILESYSTEM_SIZE));
+        } catch (Error e) {
+            critical (e.message);
+            hdd = _("Unknown");
+        }
+
+        try {
+            var oem_file = new KeyFile ();
+            oem_file.load_from_file ("/etc/oem.conf", KeyFileFlags.NONE);
+            // Assume we get the manufacturer name
+            manufacturer_name = oem_file.get_string ("OEM", "Manufacturer");
+
+            // We need to check if the key is here because get_string throws an error if the key isn't available.
+            if (oem_file.has_key ("OEM", "Product")) {
+                product_name = oem_file.get_string ("OEM", "Product");
+            }
+
+            if (oem_file.has_key ("OEM", "Version")) {
+                product_version = oem_file.get_string ("OEM", "Version");
+            }
+
+            if (oem_file.has_key ("OEM", "Logo")) {
+                manufacturer_icon_path = oem_file.get_string ("OEM", "Logo");
+            }
+
+            if (oem_file.has_key ("OEM", "URL")) {
+                manufacturer_support_url = oem_file.get_string ("OEM", "URL");
+            }
+
+            oem_enabled = true;
+        } catch (Error e) {
+            debug (e.message);
+            oem_enabled = false;
+        }
+    }
+
+    private string get_graphics_from_string(string graphics) {
+        //at this line we have the correct line of lspci
+        //as the line has now the form of "00:01.0 VGA compatible controller:Info"
+        //and we want the <Info> part, we split with ":" and get the 3rd part
+        string[] parts = graphics.split(":");
+        string result = graphics;
+        if (parts.length == 3)
+            result = parts[2];
+        else if (parts.length > 3) {
```

Comment:
Since you're touching the code anyways, you should fix the cuddled braces (And add some to that if up there) :)